src/Common/src/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -259,6 +259,31 @@ public override MethodDesc GetStaticConstructor()
             return null;
         }
 
+        public override MethodDesc GetFinalizer()
+        {
+            if (!HasBaseType)
+                return null;
+
+            MetadataType objectType = Context.GetWellKnownType(WellKnownType.Object);
+
+            foreach (MethodDesc decl in VirtualFunctionResolution.EnumAllVirtualSlots(objectType))
+            {
+                if (decl.Name == "Finalize")
+                {
+                    MethodDesc impl = VirtualFunctionResolution.FindVirtualFunctionTargetMethodOnObjectType(decl, this);
+                    if (impl.OwningType != objectType)
+                    {
+                        return impl;
+                    }
+
+                    return null;
+                }
+            }
+
+            Debug.Assert(false, "Finalize method not found on System.Object");
```

Comment:
Can you do:
```csharp
    // TODO: Better exception type. Should be: "CoreLib doesn't have a required thing in it".
    throw new NotImplementedException();
```
We already have a bunch of TODOs with this message around. I'll do the right thing for them at some point.